### PR TITLE
Add toHaveText assertion

### PR DESCRIPTION
**Pull request by tronroy**
Hello! I took a stab at adding the toHaveText assertion as suggested in issue #23. Hope this helps.

### CODE REVIEW

**Comment by blainekasten** [comment]
This is awesome! Thanks so much for doing this. There was one other case I was thinking of supporting.

I was thinking if an argument wasn't provided, then it would just check for the presence of text. Maybe I don't care much about what the text is, but just that there is text? Is that worth doing?

Something like

```js
const wrapper = mount(<Fixture />);

expect(wrapper.find('p')).toHaveText(); // truthy
expect(wrapper.find('div')).toHaveText(); // falsy
```

Is that encouraging a bad testing habit? Should devs always know what text will be put inside their components that we can make it necessary?

**Comment by tronroy** [comment]
Thats a good question. Personally I like to be explicit as possible when testing, but I could totally see situations where I'd just want to check for the presence of text.

**Comment by blainekasten** [comment]
Definitely. I say we support it. Doesn't really hurt too much and can make some tests simpler (which is the point of this lib). Would you be willing to update your code to support that?

**Comment by tronroy** [comment]
I agree, sound great! I will update the pull request as soon as I can.

File: src/assertions/toHaveText.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export default {
+  toHaveText() : Object {
+    return {
+      compare(enzymeWrapper:Object, text:string) : Object {
+        const actualText = enzymeWrapper.text();
+
+        return {
+          pass: actualText === text,
+          message: `Expected "${actualText}" to equal ${text}`,
```
Comment (blainekasten) [inline src/assertions/toHaveText.js]: should wrap `${text}` in quotes also

File: src/assertions/toHaveText.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * LICENSE file in the root directory of this source tree. *
+ *
+ * @providesModule toHaveText
+ * @flow
+ */
+
+export default {
+  toHaveText() : Object {
+    return {
+      compare(enzymeWrapper:Object, text:string) : Object {
```
Comment (blainekasten) [inline src/assertions/toHaveText.js]: the flowtype tag here should be optional since we're permitting undefined here. 

```
text:?string
```

**Comment by blainekasten** [comment]
@tronroy this is looking awesome! One last fix and i'll merge it!

**Comment by tronroy** [comment]
@blainekasten Fixed it!

**Comment by blainekasten** [comment]
Love it! Thanks so much!

**Comment by whawker** [comment]
@blainekasten Can you make a new jasmine-enzyme release so we can use this matcher?

**Comment by whawker** [comment]
Thanks!

**Pull request merged**